### ISSUE TITLE
folder names no longer listed?

### ISSUE THREAD

**Issue by wiicode** [nolanlawson/pretty-s3-index-html#5]
When using this file, folder names in buckets are left blank (empty).  Only files show.  Also, everything appears under index.html in branches.

**Comment by blmoore** [comment]
Fixed by https://github.com/nolanlawson/pretty-s3-index-html/issues/3#issuecomment-204241310 for me

**Comment by nolanlawson** [comment]
This project is archived, so I am closing this issue. Thanks!

**Comment by jethrolam** [comment]
This project is archived but for anyone who's interested in this topic:

The demo HTML [https://nolanlawson.s3.amazonaws.com/index.html] seemingly displays both files and folders, but the script never automatically discovers the folders.  The folders were displayed because they are artificially created as extra objects with keys ending in `$folder$`, e.g.,:

    <Key>clockwork/mp3_$folder$</Key>

This enables the script to interpret and display this file as folder.  See the original XML [https://nolanlawson.s3.amazonaws.com/]

In summary, if you want to display folders, you need to create these extra objects.

**Comment by pmithil7** [comment]
Hi, @nolanlawson @jethrolam @blmoore

When i used the index.html, my website shows with just the title Index of bucket with empty columns of Name, Date Modified, Size and Type. I don't see any files or folders from my s3 bucket as well. I have set all the public permissions still i fail to see any files or folders. What could be wrong?

My bucket policy is:
{
    "Version": "2008-10-17",
    "Statement": [
        {
            "Sid": "AllowPublicRead",
            "Effect": "Allow",
            "Principal": {
                "AWS": "*"
            },
            "Action": "s3:GetObject",
            "Resource": "arn:aws:s3:::engops.openswitch.net/*"
        }
    ]
}

All my files are also open for everyone to read. But still none of the files or folders get populated. What could be wrong?

Appreciate anyone's help here.

Thanks,
Mithil

**Comment by qoomon** [comment]
I've build a new s3 browser `index.html` **[Demo](https://qoomon.github.io/aws-s3-bucket-browser/?bucket=https://s3.eu-west-1.amazonaws.com/data.openspending.org)**
